hypershift-operator/controllers/platform/aws/controller.go

```diff
@@ -588,15 +588,16 @@ func (r *AWSEndpointServiceReconciler) rejectVpcEndpointConnections(ctx context.
 			ServiceId:      aws.String(serviceID),
 			VpcEndpointIds: existingEndpointIDs,
 		}); rejectEndpointsErr != nil {
-			return unwrapError(rejectEndpointsErr)
+			return unwrapError(log, rejectEndpointsErr)
 		}
 	}
 
 	return nil
 }
 
-func unwrapError(err error) error {
+func unwrapError(log logr.Logger, err error) error {
 	if awsErr, ok := err.(awserr.Error); ok {
+		log.Info("AWS Error", "code", awsErr.Code(), "message", awsErr.Message())
```

Comment:
Logging the error message instead of returning it because of feedback I received:

> the reason we don't include the error message in the error we return is that AWS usually puts something in there that changes every time. And we update the HostedCluster status with the text from the error we return. That means that if we're constantly getting an error, we're going to be updating the HC status many times, which is not a good thing.

cc @csrwng